diff --git a/nspr/pr/tests/testfile.c_ b/nspr/pr/tests/testfile.c
index 79d8a05..3fa0ae3 100644
--- a/nspr/pr/tests/testfile.c_
+++ b/nspr/pr/tests/testfile.c
@@ -123,7 +123,7 @@ PRThread* create_new_thread(PRThreadType type,
     if (native_thread) {
 #if defined(_PR_PTHREADS)
         pthread_t tid;
-        if (!pthread_create(&tid, NULL, start, arg)) {
+        if (!pthread_create(&tid, NULL, (void*)start, arg)) {
             return((PRThread *) tid);
         }
         else {
@@ -573,7 +573,7 @@ static PRInt32 RunDirTest(void)
 
         thrarg.done= 0;
         t = create_new_thread(PR_USER_THREAD,
-                              DirTest, &thrarg,
+                              (void*)DirTest, (void*)&thrarg,
                               PR_PRIORITY_NORMAL,
                               PR_LOCAL_THREAD,
                               PR_UNJOINABLE_THREAD,
